### PR TITLE
Grid layout site docs

### PR DESCRIPTION
**Pull request by JoanaMMoreira**
Adds the `GridLayout` docs and examples to the site

Preview at: https://saltdesignsystem-git-grid-layout-docs-fed-team.vercel.app/salt/components/grid-layout/examples

### CODE REVIEW

File: site/docs/components/grid-layout/examples.mdx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  $ref: ./#/data
+---
+
+<LivePreviewControls>
+
+<LivePreview componentName="grid-layout" exampleName="Default">
+
+### Default
+
+Grid Layout defaults to a 12 column grid with one row. The spacing between rows and columns is driven by the prop [gap](https://www.saltdesignsystem.com/salt/foundations/spacing) and is set to 3.
```
Comment (alycrys) [inline site/docs/components/grid-layout/examples.mdx:18]: don't think `gap` should be a link

Comment (feliciackh) [inline site/docs/components/grid-layout/examples.mdx:18]: nit
```suggestion
Grid Layout defaults to a 12 column grid with one row. The spacing between rows and columns is driven by the prop [gap](/salt/foundations/spacing) and is set to 3.
```

File: site/docs/components/grid-layout/examples.mdx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+</LivePreview>
+
+<LivePreview componentName="grid-layout" exampleName="ResponsiveLayout">
+
+### Responsive layout
+
+The Grid Layout’s responsive properties allow you to customize specific properties of the grid (such as the number of columns and rows) per breakpoint.
+
+You can use the default [Salt breakpoints](https://www.saltdesignsystem.com/salt/foundations/responsiveness) or pass in your own using `SaltProvider`.
```
Comment (feliciackh) [inline site/docs/components/grid-layout/examples.mdx:44]: nit
```suggestion
You can use the default [Salt breakpoints](/salt/foundations/responsiveness) or pass in your own using `SaltProvider`.
```

File: site/docs/components/grid-layout/examples.mdx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+### Default
+
+Grid Layout defaults to a 12 column grid with one row. The spacing between rows and columns is driven by the prop [gap](https://www.saltdesignsystem.com/salt/foundations/spacing) and is set to 3.
+
+Content in the Grid Layout is displayed inside Grid Item components. This helps control the size and position of content.
+
+#### Guidance
+
+- Grid Item components are recommended for displaying content in the layout. If the layout is very simple (I.e., does not span multiple columns or rows) content (e.g., components, text, divs, etc) can be placed directly into the Grid Layout component.
```
Comment (alycrys) [inline site/docs/components/grid-layout/examples.mdx:24]:
```suggestion
- Grid Item components are recommended for displaying content in the layout. If the layout is very simple (i.e., does not span multiple columns or rows) content (e.g., components, text, divs, etc) can be placed directly into the Grid Layout component.
```

File: site/docs/components/grid-layout/usage.mdx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+layout: DetailComponent
+sidebar:
+  exclude: true
+data:
+  $ref: ./#/data
+---
+
+### When to use Grid Layout
+
+- Your content is best displayed using a configurable number of columns and rows, such as a [Form](https://www.saltdesignsystem.com/salt/patterns/forms) or a Dashboard.
```
Comment (feliciackh) [inline site/docs/components/grid-layout/usage.mdx:13]: nit
```suggestion
- Your content is best displayed using a configurable number of columns and rows, such as a [Form](/salt/patterns/forms) or a Dashboard.
```

File: site/docs/components/grid-layout/usage.mdx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+- For content regions where blocks, or groups, of content are displayed alongside each other, such as the main content area of the application, a Dialog, or a Drawer.
```
Comment (feliciackh) [inline site/docs/components/grid-layout/usage.mdx:15]: do we want links for Dialog and Drawer?

File: site/docs/components/grid-layout/usage.mdx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+- The application or webpage will be displayed across different viewport, screen, or device sizes. Grid Layout provides control over how the content is displayed and responsively organized across the different spaces, and ensures a reliable, usable experience.
+
+### When not to use Grid Layout
+
+- To define the high-level layout of your application surrounding the content area (i.e., header, footer, navigation areas. Instead, use the Border Layout component.
+
+- To lay out your content in a single dimension only, for example a horizontal summary bar containing a row of metric components, or a vertical stack of Accordion components. Instead, use Stack Layout, Flow Layout or Flex Layout.
```
Comment (feliciackh) [inline site/docs/components/grid-layout/usage.mdx:23]: same - links for components?

File: site/docs/components/grid-layout/examples.mdx (hunk 2 above)
Comment (alycrys) [inline site/docs/components/grid-layout/examples.mdx:44]: should SaltProvider be a link?

File: site/docs/components/grid-layout/usage.mdx (hunk 5 above)
Comment (JoanaMMoreira) [reply site/docs/components/grid-layout/usage.mdx:15]: Yeah, I was going to wait until those were on the site or should we link them to storybook?

**Review by feliciackh — APPROVED** [review]

**Review by Fercas123 — APPROVED** [review]

File: site/docs/components/grid-layout/usage.mdx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+To import Grid Layout from the core Salt package, use:
+
+```js
+import { GridLayout, GridItem } from "@salt-ds/core";
+```
+
+### Props
+
+<PropsTable packageName="core" componentName="GridLayout" />
```
Comment (origami-z) [inline site/docs/components/grid-layout/usage.mdx:35]: not all default values are shown in default column

**Review by origami-z — APPROVED** [review]

File: site/docs/components/grid-layout/usage.mdx (hunk 7 above)
Comment (JoanaMMoreira) [reply site/docs/components/grid-layout/usage.mdx:35]: (see my comment on [Flex](https://github.com/jpmorganchase/salt-ds/pull/2304))

**Pull request merged**